### DISCUSSION TITLE
How to get windows process state when process is not running

### DISCUSSION THREAD

**Discussion by kyo-hyun** [community/community#136430] [Other Feature Feedback, Questions, & Ideas]
Hi guys.
I'm a student who is studying elk stack form korea
I have deployed ELK Stack and metricbeat to check process state monitoring and log. 
But as far as I know Metricbeat only collects information from things that are there, it cannot monitor process that is not running. 
I want to check not running process at elasticsearch as a document.
so what am I going to do to check not running state windows process on elasticsearch?

**Comment by ferid333** [comment]
You're correct that Metricbeat typically collects data from running processes. To monitor a process that isn’t running and send that information to Elasticsearch, you can follow these steps:

1. **Use a Script or Custom Module:** Create a script that checks if the process is running. If the process isn’t found, the script can generate a custom log or event.

2. **Log the Event:** The script can log a message saying the process isn't running. This log can then be picked up by Filebeat or another monitoring tool.

3. **Send Data to Elasticsearch:** Once the log or event is generated, you can use Filebeat or a custom Metricbeat module to send this data to Elasticsearch as a document.

4. **Visualize the Data:** In Kibana, you can then set up alerts or visualizations to notify you when a process isn’t running.

This way, you'll be able to track both running and non-running processes in Elasticsearch.🤗

**Reply by kyo-hyun** [reply]
Thank you for your answer. 
I have solved it with the "exec" plug-in of "fluentbit".
Anyway thank you again and have a good day :)

**Reply by ferid333** [reply]
You are welcome ! Please close the discussion
